src/DataStructures/SpinWeighted.hpp

```diff
@@ -33,24 +33,45 @@ struct SpinWeighted<T, Spin, false> {
   // implicitly-defined one needs to be deleted so that the templated version
   // below doesn't suffer from ambiguous resolution
   SpinWeighted() = default;
-  SpinWeighted(SpinWeighted&) = default;
-  SpinWeighted(SpinWeighted&&) = default;
-  SpinWeighted& operator=(SpinWeighted&) = delete;
-  SpinWeighted& operator=(SpinWeighted&&) = delete;
-  ~SpinWeighted() = default;
+  SpinWeighted(const SpinWeighted&) noexcept = default;
+  SpinWeighted(SpinWeighted&&) noexcept = default;
+  SpinWeighted& operator=(const SpinWeighted&) noexcept = default;
+  SpinWeighted& operator=(SpinWeighted&&) noexcept = default;
+  ~SpinWeighted() noexcept = default;
 
+  // clang-tidy asks that these be marked explicit, but we actually do not want
+  // them explicit for use in the math operations below.
   template <typename Rhs>
-  auto& operator=(SpinWeighted<Rhs, Spin>& rhs) noexcept {
-    return data = rhs.data;
+  SpinWeighted(const SpinWeighted<Rhs, Spin>& rhs) noexcept  // NOLINT
+      : data{rhs.data} {}
+
+  template <typename Rhs>
+  SpinWeighted(SpinWeighted<Rhs, Spin>&& rhs) noexcept  // NOLINT
+      : data{std::move(rhs.data)} {}
+
+  SpinWeighted(const T& rhs) noexcept : data{rhs} {}        // NOLINT
+  SpinWeighted(T&& rhs) noexcept : data{std::move(rhs)} {}  // NOLINT
```

Comment:
Should these be explicit?  (I don't know, I just want to make sure it has been considered.)

Reply:
It's actually why they are `// NOLINT`-ed. I decided that I wanted the implicit conversion because in order to make the math operations efficient, we often end up with `SpinWeighted` wrapping the blaze expression templates, and it's more convenient to have implicit conversions for those cases.

Reply:
I am surprised that you can get useful implicit conversions with expression templates, but if you find it useful I'm fine with it.